lib/kennel.rb

```diff
@@ -45,13 +46,11 @@ class ValidationError < RuntimeError
   UpdateResult = Struct.new(:plan, :update, keyword_init: true)
 
   class Engine
-    def initialize
-      @out = $stdout
-      @err = $stderr
-      @strict_imports = true
+    def initialize(config_from = Kennel::Config.from_env, &block)
+      @config = Kennel::Config.new(config_from, &block)
```

Comment:
passing a config instance to config.new seems strange, can we pass in a config instead ?
```suggestion
    def initialize(config = Kennel::Config.from_env)
      @config = config
```

Reply:
You can pass in anything which responds to `to_h`.

Reply:
if someone needs to configure the config they can do it outside, and we can remove the `block` interface for initialize
can even make config a keyword-arg to leave room for future additions

Reply:
Yielding a mutable config object during build is a very common style that I'd like to keep.

Reply:
The most common for I see is keyword-args (also easy to build/debug etc), if things get more complicated then yielding+config object might make sense though.
Kennel is for users that don't necessarily know a lot of ruby, so keeping things simple ("pass in a hash" vs "yield this config object") is nice.